runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/AbstractWindmillStream.java

```diff
@@ -120,21 +147,23 @@ protected AbstractWindmillStream(
     this.isHealthCheckScheduled = new AtomicBoolean(false);
     this.finishLatch = new CountDownLatch(1);
     this.logger = logger;
+    this.sleeper = Sleeper.DEFAULT;
+    this.debugMetrics = StreamDebugMetrics.create();
+    this.streamTTL = streamTTL;
     this.requestObserver =
         new ResettableThrowingStreamObserver<>(
             () ->
                 streamObserverFactory.from(
                     clientFactory,
                     new AbstractWindmillStream<RequestT, ResponseT>.ResponseObserver()),
             logger);
-    this.sleeper = Sleeper.DEFAULT;
-    this.debugMetrics = StreamDebugMetrics.create();
   }
 
-  private static String createThreadName(String streamType, String backendWorkerToken) {
+  private static String createThreadName(
+      String threadType, String streamType, String backendWorkerToken) {
```

Comment:
This doesn't quite match how we're using it. It looks like above we're passing in `(streamType, backendWorkerToken, threadType)`.  It seems like we shouldn't need to hardcode "WindmillStream" in this method either, since it'll be passed in as threadType somewhere?